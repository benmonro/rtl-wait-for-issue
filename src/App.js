import React, { useState } from 'react';
import logo from './logo.svg';
import './App.css';

function App() {
  const [showStuff, setShowStuff] = useState(false);
  function handleClick() {
    setTimeout(() => {

      setShowStuff(true);
    }, 500)
  }
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <button onClick={handleClick}>click me</button>
        {showStuff && <div>Hello world</div>}
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </a>
      </header>
    </div>
  );
}

export default App;
