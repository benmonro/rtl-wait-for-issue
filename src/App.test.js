import React from 'react';
import { render, waitFor, screen, fireEvent } from '@testing-library/react';
import App from './App';

test('renders learn react link', () => {
  const { getByText } = render(<App />);
  const linkElement = getByText(/learn react/i);
  expect(linkElement).toBeInTheDocument();
});

test("should match snapshot async", async () => {
  render(<App />);
  // fireEvent.click(screen.getByText("click me"));

  // waitFor()
  expect(document.body).toMatchSnapshot();
})
