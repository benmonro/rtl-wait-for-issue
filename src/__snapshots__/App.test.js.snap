// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot async 1`] = `
Array [
  Array [
    Object {
      "foo": "bar",
    },
  ],
]
`;
